docs/how-to/enable_hugepages_for_upf.md

```diff
@@ -0,0 +1,38 @@
+# Enable HugePages in the UPF
+
+## Requirements
+
+- A Kubernetes Host 
+  - CPU that supports AVX2 and RDRAND instructions (Intel Haswell, AMD Excavator or equivalent)
+  - CPU has the flag "pdpe1gb" which allows to set 1Gi HugePages
+  - HugePages is enabled and has enough 2Gi free HugePages
+- A Kubernetes cluster with the Multus addon enabled
+- Juju = 3.1/stable
+
+## Deploy the UPF
+
+Deploy the UPF charm in a Juju model named `user-plane` with HugePages enabled.
+
+```bash
+juju add-model user-plane
+juju deploy sdcore-upf --trust --channel=edge --config enable-hugepages=True
+```
+
+## Validate HugePages support
+
+Check the bessd container logs which indicate that the HugePages are reserved.
+
+```bash
+kubectl logs -f  sdcore-upf-0 -c  bessd -n user-plane  | grep -i huge
+# Sample Output
+2023-10-26T12:33:35.495Z [bessd] I1026 12:33:35.495568    54 dpdk.cc:169] Initializing DPDK EAL with options: ["bessd", "--main-lcore", "127", "--lcore", "127@0-31", "--no-shconf", "--legacy-mem", "--socket-mem", "1024", "--huge-unlink"]
+2023-10-26T12:33:35.506Z [bessd] EAL: 2 hugepages of size 1073741824 reserved, but no mounted hugetlbfs found for that size
```

Comment:
This line is actually a warning saying that `bessd` has reserved huge pages of a size (1073741824) but they are not mounted in the expected path (`/dev/hugepages`).
I just reconfigured my host with 1Gi huge pages and my log says:
```
2023-10-27T12:34:11.552Z [bessd] I1027 12:34:11.552897    53 dpdk.cc:169] Initializing DPDK EAL with options: ["bessd", "--main-lcore", "127", "--lcore", "127@0-23", "--no-shconf", "--legacy-mem", "--socket-mem", "1024", "--huge-unlink"]
2023-10-27T12:34:11.554Z [bessd] EAL: Detected 24 lcore(s)
2023-10-27T12:34:11.554Z [bessd] EAL: Detected 1 NUMA nodes
2023-10-27T12:34:11.554Z [bessd] EAL: Detected static linkage of DPDK
2023-10-27T12:34:11.557Z [bessd] EAL: Selected IOVA mode 'VA'
2023-10-27T12:34:11.557Z [bessd] EAL: No available hugepages reported in hugepages-2048kB
2023-10-27T12:34:11.557Z [bessd] EAL: Probing VFIO support...
2023-10-27T12:34:11.720Z [bessd] EAL: No legacy callbacks, legacy socket not created
2023-10-27T12:34:11.720Z [bessd] I1027 12:34:11.719929    53 packet_pool.cc:49] Creating DpdkPacketPool for 262144 packets on node 0
2023-10-27T12:34:11.720Z [bessd] I1027 12:34:11.719936    53 packet_pool.cc:70] PacketPool0 requests for 262144 packets
2023-10-27T12:34:11.736Z [bessd] I1027 12:34:11.736119    53 packet_pool.cc:157] PacketPool0 has been created with 262144 packets
```
The line logging `[bessd] EAL: No available hugepages reported in hugepages-2048kB` is expected as I am not allocating huge pages of 2Mi on the host.

I would remove this line from here.

Reply:
Done